crates/types/src/messages.rs

```diff
@@ -1,12 +1,15 @@
+use std::net::SocketAddr;
+
 use serde::{Deserialize, Serialize};
+
 use serialize_hierarchy::SerializeHierarchy;
 use spl_network_messages::{
     GameControllerReturnMessage, GameControllerStateMessage, HulkMessage, VisualRefereeMessage,
 };
 
 #[derive(Clone, Debug, Deserialize, Serialize, SerializeHierarchy)]
 pub enum IncomingMessage {
-    GameController(GameControllerStateMessage),
+    GameController(SocketAddr, GameControllerStateMessage),
```

Comment:
Hmmm, my rule of thumb is that I try to use tuple structs only if I have a single wrapped type. You have two. Is it already unclear what the address means s.t. we should introduce proper names for the fields? On the other hand, it would be inconsistent with the remaining enum variants... 🤔

Reply:
I think its fine the way it is.